lib/Zonemaster/Backend/Translator.pm

```diff
@@ -18,9 +18,14 @@ require Zonemaster::Engine::Logger::Entry;
 extends 'Zonemaster::Engine::Translator';
 
 Readonly my %TAG_DESCRIPTIONS => (
+    TEST_DIED => sub {
+        __x    # BACKEND_TEST_AGENT:TEST_DIED
+          "An error occurred, Zonemaster could not finish the test.", @_;
+    },
     UNABLE_TO_FINISH_TEST => sub {
         __x    # BACKEND_TEST_AGENT:UNABLE_TO_FINISH_TEST
-          "Zonemaster was unable to start or finish the test.", @_;
+          'Execution took too long (the current limit is {max_execution_time} seconds). '
+          . 'Maybe too many name servers or not responsive enough.', @_;
```

Comment:
"The test took too long time to run (the current limit is {max_execution_time} seconds). Maybe there are too many name servers or the name servers are not reachable or not responsive enough."

It is easier to understand with complete sentences and translation is easier.

Reply:
updated